bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerFragmentReplicator.java

```diff
@@ -133,6 +133,16 @@ public LedgerFragmentReplicator(BookKeeper bkc, ClientConfiguration conf) {
     private static final Logger LOG = LoggerFactory
             .getLogger(LedgerFragmentReplicator.class);
 
+    private boolean isEntryExpectedOnDeadBookie(final LedgerHandle lh, Long eId, Set<Integer> deadBookieIndex) {
+        DistributionSchedule.WriteSet writeSet = lh.getWriteSetForReadOperation(eId);
```

Comment:
we should call `writeSet .recycle()` in a finally block in order to reduce allocations

Reply:
The same advice,
you needs to recycle writeSet at the end,code like 

```
try{
     DistributionSchedule.WriteSet writeSet = lh.getWriteSetForReadOperation(eId);
     for (Integer deadIndex : deadBookieIndex) {
            if (!writeSet.contains(deadIndex)) {
                return false;
            }
      }
      writeSet.recycle();
}
```